src/SchemaRegistry.ts

```diff
@@ -158,6 +164,62 @@ export default class SchemaRegistry {
     return registeredSchema
   }
 
+  private async updateOptionsWithSchemaReferences(
+    options: SchemaRegistryAPIClientOptions | undefined,
+    schema: ConfluentSchema,
+  ) {
+    const helper = helperTypeFromSchemaType(schema.type)
+    const referredSchemas = await this.getReferredSchemas(schema, helper)
+    return helper.updateOptionsFromSchemaReferences(options as ProtocolOptions, referredSchemas)
```

Comment:
I don't really understand the casting of `options` to `ProtocolOptions` here. I don't think we know that `options` isn't `LegacyOptions` at this point, since the type hasn't been narrowed down to ProtocolOptions.

I think you have to do something similar to https://github.com/kafkajs/confluent-schema-registry/blob/05a36130d74e6bd4f86d2076306b52dcb86e696e/src/schemaTypeResolver.ts#L69-L92 to make sure that if you initialized the client with `LegacyOptions` those options are used when the schema is an Avro schema. Right now I think the `updateOptions` method is assuming that the options are always `ProtocolOptions`, which would break backwards compatibility.

Reply:
The tests in SchemaRegistry.avro.spec.ts look like they are using the LegacyOptions, though, so I'm confused how this is working. I'm kinda suspecting that we end up making a frankenstein options object here by cominbing `LegacyOptions` with `ProtocolOptions`, into something like:

```js
{
  subject: 'subject',
  [SchemaType.JSON]: {
    references: [ ... ]
  }
}
```

Reply:
I think this is a good catch, I cannot find a test that use LegacyOptions

Reply:
😔 You're right. I could have sworn there were, but apparently not. You can see the difference [here](https://kafkajs.github.io/confluent-schema-registry/docs/v2#configuring-serialization-libraries). It's a fairly minor difference, in that the options have just moved in one level to accommodate other schema types.